### ISSUE TITLE
JS Folder is empty

### ISSUE THREAD

**Issue by Memz-idk** [cincheo/jsweet-eclipse-plugin#29]
So, I followed the http://www.jsweet.org/eclipse-plugin/ tutorial but the js folder is empty, the .jsweet folder was created but I checked the .jsweet/candies/src and it was empty(Idk if it's supposed to be empty but probably not), I tested it on a dynamic web project but even on a normal project project it doesn't work, everything else seems to be working

**Comment by lgrignon** [comment]
Hello @Memz-idk 
Thanks for reporting. Which version of Eclipse, Java are you using?

**Comment by Memz-idk** [comment]
> Hello @Memz-idk
> Thanks for reporting. Which version of Eclipse, Java are you using?

I'm using 2020-03 (4.15.0)

**Comment by lgrignon** [comment]
Java version please :)

**Comment by Memz-idk** [comment]
> Java version please :)

As far as I tested it doesn't matter that much but i'm using 1.8.0_231

**Comment by lgrignon** [comment]
It's kind of important because Eclipse plugin is not compatible with Java 9+ :)

Do you launch Eclipse with JDK or JRE. Please try JDK if you launch it with JRE.

Do you have any error in Eclipse error log?

Thanks

**Comment by Memz-idk** [comment]
> It's kind of important because Eclipse plugin is not compatible with Java 9+ :)
> 
> Do you launch Eclipse with JDK or JRE. Please try JDK if you launch it with JRE.
> 
> Do you have any error in Eclipse error log?
> 
> Thanks

Ok I tested again and I found this error in the logs
`
!ENTRY org.jsweet.plugin 4 0 2020-09-08 11:27:32.027
!MESSAGE cannot compile
!STACK 0
java.lang.NoSuchFieldError: CLASSPATH
	at org.jsweet.transpiler.JSweetTranspiler.initJavac(JSweetTranspiler.java:610)
	at org.jsweet.transpiler.JSweetTranspiler.setupCompiler(JSweetTranspiler.java:736)
	at org.jsweet.transpiler.JSweetTranspiler.java2ts(JSweetTranspiler.java:882)
	at org.jsweet.transpiler.JSweetTranspiler.transpile(JSweetTranspiler.java:835)
	at org.jsweet.plugin.builder.JSweetBuilder.transpileFiles(JSweetBuilder.java:543)
	at org.jsweet.plugin.builder.JSweetBuilder.fullBuild(JSweetBuilder.java:586)
	at org.jsweet.plugin.builder.JSweetBuilder.build(JSweetBuilder.java:488)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:832)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:220)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:316)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:319)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:371)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:392)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:154)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:244)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
`

**Comment by lgrignon** [comment]
It is not compatible with java over 8, please try with jdk 8 and post here your error if any

**Comment by Memz-idk** [comment]
> It is not compatible with java over 8, please try with jdk 8 and post here your error if any

Same error :/

**Comment by lgrignon** [comment]
I strongly recommend you to use https://github.com/lgrignon/jsweet-maven-plugin which is up to date with latest 3.0.0-RC8
(but also 2.3.10 if you want to stick with Java 8).
Maven plugin can be integrated in Eclipse, and there is a watch mojo which is pretty handful.

Anyway, it would be great to refresh Eclipse JSweet plugin, even if it's not a priority. For instance, it could use the latest JSweet version and ensure it supports recent Eclipse versions.


Another important point for you if you want to try again with current plugin: 
- did you use openjdk? JSweet 2 requires Oracle JDK if my memory is correct
- candies directory can be empty if you do not reference any candy in your project's dependencies ([candies are JSweet jars](http://www.jsweet.org/faq/#How_does_JSweet_work))

**Comment by ragaller** [comment]
I also followed the plugin tutorial as mentioned in the initial comment and same error here:

```
eclipse.buildId=4.19.0.I20210303-1800
java.version=15.0.2
java.vendor=Oracle Corporation
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=de_DE
Framework arguments:  -product org.eclipse.epp.package.java.product
Command-line arguments:  -os linux -ws gtk -arch x86_64 -product org.eclipse.epp.package.java.product

org.jsweet.plugin
Error
Sun May 16 16:35:37 CEST 2021
cannot compile

java.lang.NoSuchFieldError: CLASSPATH
	at org.jsweet.transpiler.JSweetTranspiler.initJavac(JSweetTranspiler.java:610)
	at org.jsweet.transpiler.JSweetTranspiler.setupCompiler(JSweetTranspiler.java:736)
	at org.jsweet.transpiler.JSweetTranspiler.java2ts(JSweetTranspiler.java:882)
	at org.jsweet.transpiler.JSweetTranspiler.transpile(JSweetTranspiler.java:835)
	at org.jsweet.plugin.builder.JSweetBuilder.transpileFiles(JSweetBuilder.java:543)
	at org.jsweet.plugin.builder.JSweetBuilder.fullBuild(JSweetBuilder.java:586)
	at org.jsweet.plugin.builder.JSweetBuilder.build(JSweetBuilder.java:488)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:846)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:229)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:277)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:330)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:333)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:385)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:406)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:154)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:244)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

Digging into the https://github.com/cincheo/jsweet-eclipse-plugin repository I realized that the plugin installed from the update site is pretty old (2019-04-22), since http://eclipse-update-site.jsweet.org/site.xml points to version `0.9.6.20190422`, although there is a more recent commit https://github.com/cincheo/jsweet-eclipse-plugin/commit/5aac78da9471f2665b299836a485d35dbf1c6b54 which includes an almost up-to-date transpiler.

Thus I built the plugin myself from the above mentioned commit and lo and behold it works.

I'm running the `eclipse-java-2021-03-R-linux-gtk-x86_64.tar.gz` with the prepackaged jre (`plugins/org.eclipse.justj.openjdk.hotspot.jre.full.linux.x86_64_15.0.2.v20210201-0955/jre/bin/java`) on Debian Buster.

My sample jsweet project itself uses the default jre from Debian:

```
openjdk 11.0.11 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9-post-Debian-1deb10u1)
OpenJDK 64-Bit Server VM (build 11.0.11+9-post-Debian-1deb10u1, mixed mode, sharing)
```

Long story short: It's probably enough to build a plugin release with the new transpiler and publish it to the update site.

**Comment by minchu125** [comment]
I tried to install JSweet plugin for eclipse,  Luna Service Release 2 (4.4.2), got following error:
HTTP Server 'Internal Error': http://eclipse-update-site.jsweet.org/content.xml
HTTP Server 'Internal Error': http://eclipse-update-site.jsweet.org/content.xml
HttpComponents connection error response code 500.
Please help!

**Comment by renaudpawlak** [comment]
Service is temporary down for SSL migration. Sorry for the inconvenience but unfortunately it is complicated to migrate apparently.

**Comment by lgrignon** [comment]
Hello @minchu125 
Is it better for you now? You can also try to clear DNS cache. It should be fixed right now.

Please close if it's ok
